### PR TITLE
fix: low version of mongo not support retryWrites setting.

### PR DESCRIPTION
**Pull request by onebite**
Motivation:

provide a way to set retryable settings

### CODE REVIEW

File: src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java [hunk 1]
```diff
@@ -10,14 +10,20 @@
 import org.bson.codecs.configuration.CodecRegistries;
 import org.bson.codecs.configuration.CodecRegistry;
 
+import java.util.HashSet;
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java:13]:
```suggestion
```

File: src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java [hunk 2]
```diff
@@ -10,14 +10,20 @@
...
 import java.util.List;
 import java.util.Objects;
+import java.util.Set;
+import static java.util.Arrays.asList;
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java:17]:
```suggestion
```

File: src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java [hunk 3]
```diff
@@ -10,14 +10,20 @@
...
 
 /**
  * @author <a href="mailto:nscavell@redhat.com">Nick Scavelli</a>
  */
 public class MongoClientOptionsParser {
 
+  private static final Set<String> TRUE_VALUES = new HashSet<String>(asList("true", "yes", "1"));
+  private static final Set<String> FALSE_VALUES = new HashSet<String>(asList("false", "no", "0"));
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java:25]:
```suggestion
```

File: src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java [hunk 4]
```diff
@@ -84,9 +90,37 @@ public MongoClientOptionsParser(Vertx vertx, JsonObject config) {
...
     this.settings = options.build();
   }
 
+  public void applyRetryableSetting(MongoClientSettings.Builder options,ConnectionString cs,JsonObject config){
+    if(cs != null){
+      options.retryWrites(cs.getRetryWritesValue());
+      options.retryReads(cs.getRetryReads());
+    }else {
+      Boolean retryWrites = parseBoolean(config.getString("retryWrites"));
+      Boolean retryReads = parseBoolean(config.getString("retryReads"));
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java:105]:
```suggestion
      Boolean retryWrites = config.getBoolean("retryWrites");
      Boolean retryReads = config.getBoolean("retryReads");
```

File: src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java [hunk 5]
```diff
@@ -84,9 +90,37 @@ public MongoClientOptionsParser(Vertx vertx, JsonObject config) {
...
+  }
+
+  private Boolean parseBoolean(String value){
+    if(value == null){
+      return null;
+    }
+    String trim = value.toLowerCase().trim();
+
+    return TRUE_VALUES.contains(trim) || !FALSE_VALUES.contains(trim);
+  }
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java:122]:
```suggestion
```

**Review by tsegismont — APPROVED** [review]
Thanks for the updates.

It looks good to me, except for the boolean value parsing. When the config comes from a `JsonObject`, we'll stick to "standard" JSON boolean values.

**Comment by onebite** [comment]
thank you for your patience. all suggestions have submit.

**Comment by tsegismont** [comment]
You're welcome. I'm waiting for the CI job result before merging.

**Pull request merged**